### DISCUSSION TITLE
Azure storage

### DISCUSSION THREAD

**Discussion by RJanssens** [structurizr/onpremises#92] [General]
My organisation uses Azure BLOB storage rather than S3-compatible storage. I adapted the code to support this as an alternative storage provider. If there's interest, let me know and I'll create a pull request to add it to the codebase.

**Comment by maldworth** [comment]
This would be nice, and I would have used it if it was available when I was building out our infrastructure for the on-premesis. But since I deployed it to an Azure Container App, I used the storage `file` configuration, and just mounted an Azure File Share into the Container Image, and it works all the same.

**Comment by PerArneMunthe** [comment]
> My organisation uses Azure BLOB storage rather than S3-compatible storage. I adapted the code to support this as an alternative storage provider. If there's interest, let me know and I'll create a pull request to add it to the codebase.

Could you please share what you had to do ? I'm in the same situation with Azure Blob storage as the only option

**Comment by simonbrowndotje** [comment]
An initial implementation for Azure Blob Storage has been added to the codebase -> https://docs.structurizr.com/onpremises/data-storage#azure-blob-storage

Just as an FYI though, you will see better performance mounting a persistent volume and using file based storage, particularly once you get to tens or hundreds of workspaces on a single installation. This also applies to AWS S3.

**Comment by RJanssens** [comment]
Thanks Simon! We're using Structurizr a bit differently now. We store our model (software systems, relations, persons etc) and the views in a back-end so entities only need to be defined once. Views are translated to the DSL by calling the StructurizrDslParser in the back-end and then rendered on-the-fly in the front-end using the JavaScript StructurizrDiagram.
